config/routes.rb

```diff
@@ -21,11 +21,15 @@
   resources :trainees, only: %i[index show new create update] do
     scope module: :trainees do
       resource :contact_details, concerns: :confirmable, only: %i[edit update], path: "/contact-details"
-      resources :degrees
       namespace :degrees do
         get "/new/type", to: "type#new"
         post "/new/type", to: "type#create"
       end
+      resources :degrees do
```

Comment:
This should use the confirmable concern a la contact details.

Reply:
@defong and I had a fantastic time coming up with these creative paths. 

We couldn't get them into a suitable reusable state.

Reply:
Suggestions welcome...

Reply:
We're defining two route segments both with `degrees` ie via `resources` and `namespace`. What are the routes we're trying to achieve here? I think we should try to keep things consistent with the other sections if possible :)